spec/swagger_helper.rb

```diff
@@ -200,7 +213,9 @@
             }
           }
         }
-      }
+      },
+      createPayment: { '$ref': '#/definitions/payment' },
```

Comment:
One quick note; all those fields shouldn't be required on update; wonder if there's a way to DRY this up and only apply "required" to the create action 🤔

Reply:
this is where my Swagger knowledge is lacking.
Does that mean that only the 
```
 amount: { type: :integer, minimum: 0, example: 19_999 },
            care_finished_on: { type: :string, example: '2020-03-15' },
            care_started_on: { type: :string, example: '2020-01-01' },
            discrepancy: { type: :integer, example: 456 },
            paid_on: { type: :string, example: '2020-05-20' },
```
should be in the list of fields for an update?

You can refer to something you've previously defined with a  `'$ref': '...'` statement.   It just does a literal substitution (plops whatever the thing is right there at your `'$ref'...` statement.
Perhaps you knew that already. :-)

Reply:
Ya!  So, look at the `createUser` and `updateUser` keys; `createUser` has a `required` key nested inside with an array of fields, but `updateUser` *doesn't*.  That's because the required fields are only required when creating a new record; when updating, all the fields are optional.  So if you use your `payment` definition for both `create` and `update` it'll mark those fields as required for `update` too.